src/components/MainBlock.tsx

```diff
@@ -44,155 +28,59 @@ const container = classnames(
-  baseFontSize,
-  textColor('text-primary-light'),
-  textAlign('text-center')
-)
-
-const textArea = (isValid: boolean) =>
-  classnames(
-    baseFontSize,
-    padding('pt-6'),
-    backgroundColor('bg-transparent'),
-    resize('resize-none'),
-    textColor(isValid ? 'text-white' : 'text-gray-300'),
-    placeholderColor('placeholder-gray-400'),
-    placeholderOpacity('placeholder-opacity-70'),
-    textAlign('text-center'),
-    outlineColor('outline-transparent'),
-    transitionProperty('transition-colors')
-  )
-
-const pasteButton = (active: boolean) =>
-  classnames(
-    fontSize('text-base'),
-    backgroundColor('bg-transparent'),
-    borderWidth('border-2'),
-    borderColor(active ? 'border-gray-400' : 'border-white'),
-    textColor(active ? 'text-gray-400' : 'text-white'),
-    borderRadius('rounded-full'),
-    padding('py-2', 'px-4')
-  )
-
-const letsGoButton = classnames(
-  display('flex'),
-  justifyContent('justify-center'),
-  alignItems('items-center'),
-  gap('gap-x-3'),
-  fontSize('text-lg'),
-  backgroundColor('bg-night'),
-  borderRadius('rounded-full'),
-  padding('py-4', 'px-6'),
-  width('w-full')
-)
-const errorText = classnames(
-  textColor('text-red-500'),
-  textAlign('text-center')
-)
-
 export default function MainBlock() {
-  const [token, setToken] = useState('')
-  const [paste, setPaste] = useState(false)
-  const [pastingAttempts, setPastingAttempts] = useState(0)
-  const [validToken, setValidToken] = useState(false)
-
-  const { createProof, error, loading, setError, setLoading } = useProof()
+  const { createAttestation, createPassword } = useProof()
 
   const onMessage = useCallback(
     async (message: Message) => {
-      switch (message.type) {
-        case MessageType.CreateProof:
-          await createProof(message.params, message.signature)
-          setPastingAttempts(0)
-          if (message.params.token) {
-            onChangeText(message.params.token)
-          }
-          break
-        case MessageType.Reset:
-          setToken('')
-          setLoading(false)
-          break
-        case MessageType.Status:
-          postWebViewMessage({
-            data: {},
-            type: Messages.Ready,
-          })
-          break
-        case MessageType.Error:
-          setToken('')
-          setError(`Can't generate proof. Please try again!`)
-          setLoading(false)
-          break
+      try {
+        switch (message.type) {
+          case MessageType.CreateAttestationProof:
+            if (
+              !isValidAttestationProofMessage(message.params) ||
+              !message.signature
+            )
+              throw new Error('Invalid data for attestation proof!')
+            await createAttestation(message.params, message.signature)
+            break
+          case MessageType.CreatePasswordProof:
+            if (!isValidPasswordProofMessage(message.params))
+              throw new Error('Invalid data for password proof!')
```

Comment:
Should we use `handleError` here to log issues to the Sentry?

Reply:
There's not `Sentry` here, we'll catch this error on frontend